ci/buildspec_main.yml

```diff
@@ -0,0 +1,28 @@
+# AWS CodeBuild config
+# This file is referenced in the infra repo as a CodeBuild buildspec target.
+
+# These 3 variables are set in the codebuild project definition's environment secion.
+# ECR_REGION
+# ECR_ACCOUNT_ID
+# ECR_NAMESPACE
+
+version: 0.2
+env:
+  shell: bash
+phases:
+  install:
+    runtime-versions:
+      nodejs: 16
+  pre_build:
+    commands:
+      - aws ecr get-login-password --region "${ECR_REGION}" | docker login --username AWS --password-stdin "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com"
+
+  build:
+    commands:
+      # Backend
+      - docker build backend/core -f build/docker/Dockerfile.backend --target tests
+      - docker build backend/core -f build/docker/Dockerfile.backend --target run
+      - docker tag $(docker images -q --format='{{.ID}}' | head -1) "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/backend:${CODEBUILD_RESOLVED_SOURCE_VERSION:0:8}"
+      - docker tag $(docker images -q --format='{{.ID}}' | head -1) "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/backend:latest"
+      - docker push "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/backend:${CODEBUILD_RESOLVED_SOURCE_VERSION:0:8}"
```

Comment:
Verify this, but I'm pretty sure you don't need to explicitly push to ECR from CodeBuild.  Just add the image(s) to the `artifacts` section and CodeBuild will take care of the rest as long as you have the ECR repo defined in the build project.

```
artifacts:
  files:
    - doorway/backend:run
```

Reply:
I'm not seeing this in the [buildspec docs](https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html)